sql/moz-fx-data-shared-prod/telemetry_derived/clients_summary_v1/query.sql

```diff
@@ -0,0 +1,352 @@
+  SAFE_CAST(environment.system.os.windows_ubr AS INT64) AS windows_ubr,
+  SAFE_CAST(environment.system.os.install_year AS INT64) AS install_year,
+  environment.system.is_wow64,
+  SAFE_CAST(environment.system.memory_mb AS INT64) AS memory_mb,
+  environment.system.cpu.count AS cpu_count,
+  environment.system.cpu.cores AS cpu_cores,
+  environment.system.cpu.vendor AS cpu_vendor,
+  environment.system.cpu.family AS cpu_family,
+  environment.system.cpu.model AS cpu_model,
+  environment.system.cpu.stepping AS cpu_stepping,
+  SAFE_CAST(environment.system.cpu.l2cache_kb AS INT64) AS cpu_l2_cache_kb,
+  SAFE_CAST(environment.system.cpu.l3cache_kb AS INT64) AS cpu_l3_cache_kb,
+  SAFE_CAST(environment.system.cpu.speed_m_hz AS INT64) AS cpu_speed_mhz,
+  environment.system.gfx.features.d3d11.status AS gfx_features_d3d11_status,
+  environment.system.gfx.features.d2d.status AS gfx_features_d2d_status,
+  environment.system.gfx.features.gpu_process.status AS gfx_features_gpu_process_status,
+  environment.system.gfx.features.advanced_layers.status AS gfx_features_advanced_layers_status,
+  SAFE_CAST(environment.profile.creation_date AS INT64) AS profile_creation_date,
+  payload.info.previous_build_id,
+  payload.info.subsession_start_date,
+  payload.info.subsession_counter,
+  payload.info.subsession_length,
+  environment.partner.distribution_id,
+  IFNULL(
+    environment.services.account_enabled,
+    udf.boolean_histogram_to_boolean(payload.histograms.fxa_configured)
+  ) AS fxa_configured,
+  IFNULL(
+    environment.services.sync_enabled,
+    udf.boolean_histogram_to_boolean(payload.histograms.weave_configured)
+  ) AS sync_configured,
+  udf.histogram_max_key_with_nonzero_value(
+    payload.histograms.weave_device_count_desktop
+  ) AS sync_count_desktop,
+  udf.histogram_max_key_with_nonzero_value(
+    payload.histograms.weave_device_count_mobile
+  ) AS sync_count_mobile,
+  application.build_id AS app_build_id,
+  application.display_version AS app_display_version,
+  application.name AS app_name,
+  application.version AS app_version,
+  environment.build.build_id AS env_build_id,
+  environment.build.version AS env_build_version,
+  environment.build.architecture AS env_build_arch,
+  environment.settings.e10s_enabled,
+  environment.settings.locale,
+  environment.settings.update.channel AS update_channel,
+  environment.settings.update.enabled AS update_enabled,
+  environment.settings.update.auto_download AS update_auto_download,
+  STRUCT(
+    environment.settings.attribution.source,
+    environment.settings.attribution.medium,
+    environment.settings.attribution.campaign,
+    environment.settings.attribution.content
+  ) AS attribution,
+  environment.settings.sandbox.effective_content_process_level AS sandbox_effective_content_process_level,
+  payload.info.timezone_offset,
+  udf.extract_histogram_sum(
+    mozfun.map.get_key(payload.keyed_histograms.subprocess_crashes_with_dump, 'pluginhang')
+  ) AS plugin_hangs,
```

Comment:
Have you tried collecting all these `extract_histogram_sum/get_key` calls into a single call on top of an array like count histograms? In this case, I think we would need to make an array of STRUCT<histogram, key>.

Reply:
I'm basically wondering if we could still have a path to avoiding clients_summary by fully buying into the strategy of packing arrays for all UDF calls.

Reply:
> I'm basically wondering if we could still have a path to avoiding clients_summary by fully buying into the strategy of packing arrays for all UDF calls.

I did attempt this, and it still exceeded query complexity limits